### ISSUE TITLE
angular library scss - Could not resolve "/assets/image/image.png"

### ISSUE THREAD

**Issue by Squixx** [nrwl/nx#21609]
### Current Behavior

when compiling a angular library using @nx/angular:packager (or ng-packager-lite)  i get the error:

```
------------------------------------------------------------------------------
Building entry point '@demo/url-scss'
------------------------------------------------------------------------------
⠇ Compiling with Angular sources in Ivy partial compilation mode.✘ [ERROR] Could not resolve "/assets/image/image.png"

    <stdin>:2:24:
      2 │   background-image: url("/assets/image/image.png");
        ╵                         ~~~~~~~~~~~~~~~~~~~~~~~~~

✖ Compiling with Angular sources in Ivy partial compilation mode.

```

while in any nx version <17.3 this was working just fine

When updateing from 17.2.8 to any higher version libraries using background-image: url() refuse to compile with this error.

Note: i'm only updating NX in this case

### Expected Behavior

the lib compiles

### GitHub Repo

https://github.com/Squixx/nx-url-scss

### Steps to Reproduce

1.
```
 nx run url-scss:build:production
```

### Nx Report

```shell
nx report            

 >  NX   Report complete - copy this into the issue template

   Node   : 20.10.0
   OS     : linux-x64
   npm    : 10.2.3
   
   nx (global)        : 18.0.1
   nx                 : 18.0.1
   @nx/js             : 18.0.1
   @nx/jest           : 18.0.1
   @nx/linter         : 18.0.1
   @nx/eslint         : 18.0.1
   @nx/workspace      : 18.0.1
   @nx/angular        : 18.0.1
   @nx/cypress        : 18.0.1
   @nx/devkit         : 18.0.1
   @nx/eslint-plugin  : 18.0.1
   @nrwl/tao          : 18.0.1
   @nx/web            : 18.0.1
   @nx/webpack        : 18.0.1
   typescript         : 5.3.3
```

### Failure Logs

```shell
nx run url-scss:build                                   

> nx run url-scss:build:production

Building Angular Package

------------------------------------------------------------------------------
Building entry point '@demo/url-scss'
------------------------------------------------------------------------------
⠇ Compiling with Angular sources in Ivy partial compilation mode.✘ [ERROR] Could not resolve "/assets/image/image.png"

    <stdin>:2:24:
      2 │   background-image: url("/assets/image/image.png");
        ╵                         ~~~~~~~~~~~~~~~~~~~~~~~~~

✖ Compiling with Angular sources in Ivy partial compilation mode.

 >  NX   Build failed with 1 error:

   <stdin>:2:24: ERROR: Could not resolve "/assets/image/image.png"
   Pass --verbose to see the stacktrace.
```

### Package Manager Version

pnpm, npm

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by leosvelperez** [comment]
Thanks for reporting this!

This is an issue with the upstream `ng-packagr` package, which was already addressed by https://github.com/ng-packagr/ng-packagr/pull/2783. The change is still not released. If it's released in a patch version, you'll just need to update the `ng-packagr` version in your project. If it's released as part of Angular v17.2, we'll release support for it in due time.

**Comment by Squixx** [comment]
Thanks I'll monitor it there
